src/client/WebSocketClient.ts

```diff
@@ -464,7 +478,12 @@ export class WebSocketClient extends EventEmitter {
     return this.socket;
   }
 
-  disconnect(reason: string = 'Unknown reason'): void {
+  disconnect(reason: string = 'Unknown reason', forgetSubscriptions: boolean = false): void {
```

Comment:
I'm not too thrilled about this function signature because giving just the `forgetSubcriptions` without a reason requires something like this:

```TypeScript
ws.disconnect(undefined, true);

// OR

ws.disconnect('', true);
```

Maybe another option would be to convert the parameters to an options object like so:

```TypeScript
disconnect(options?: {
  reason?: boolean,
  forgetSubscriptions?: boolean,
}): void
```

We could accomplish that without changing the API with some checks if the param is a string or an object.

Reply:
I'm sure you saw it, but I pushed a commit that changes the parameter to `options` while still allowing it to be a string indicating the disconnect reason for backward compatibility.